README.md

```diff
@@ -64,6 +64,37 @@ List of repository sources:
 - [/flow-emulator](https://github.com/onflow/flow-emulator)
 - [/flow-cadut](https://github.com/onflow/flow-cadut)
 
-## Further help
+## Deployment
 
-Visit the [Nx Documentation](https://nx.dev) to learn more.
+⚠ currently data cached in redis is not invalidated from new bbuilds, so e.g. you might see cached mdx and markdown data from previous builds. to work around this, you need to manually remove keys from redis.
+
+Commits pushed to the `main` branch are automatically deployed to staging. Commits pushed to `production` are automatically deployed to production. See [deploy.yml](.github/workflows/deploy.yml) for details.
+
+Recommended workflow to promote changes from `main` (staging) to production is:
+
+```bash
+git fetch
+git switch production
+git reset --hard origin/main
+git push --force
+```
+
+### Deployment takes forever or fails due to memory errors
+
+If deploys fail to go out, and the github actions hang indefinitely, a workaround to fix this is:
+
+1. cancel the build
+1. delete the fly app that is prefixed with `fly-builder-`, this can be done through CLI as shown below, or through the [fly dashboard](https://fly.io/dashboard/flow-docs)
```

Comment:
```suggestion
1. Delete the fly app that is prefixed with `fly-builder-`, this can be done through CLI as shown below, or through the [fly dashboard](https://fly.io/dashboard/flow-docs)
```